### PR TITLE
Implment delete user

### PR DESCRIPTION
**Pull request by Cryptex-github**
See title

### CODE REVIEW

File: ferrischat_webserver/src/users/delete_user.rs [hunk 1]
```diff
@@ -1,6 +1,33 @@
...
+            Some(_) => HttpResponse::NoContent().finish(),
+            None => HttpResponse::NotFound().json(NotFoundJson {
+                message: "User not found".to_string(),
+            }),
+        },
+        Err(e) => HttpResponse::InternalServerError().json(InternalServerErrorJson {
+            reason: format!("DB Returned a error: {}", e),
+        }),
+    }
     HttpResponse::NoContent()
```
Comment (hydrostaticcog) [inline ferrischat_webserver/src/users/delete_user.rs:32]: this is unnecessary.

**Review by hydrostaticcog — CHANGES_REQUESTED** [review]

File: ferrischat_webserver/src/users/delete_user.rs (hunk 1 above)
Comment (tazz4843) [inline ferrischat_webserver/src/users/delete_user.rs]: it looks as if it won't even compile with this line here

File: ferrischat_webserver/src/users/delete_user.rs [hunk 2]
```diff
@@ -1,6 +1,33 @@
 use actix_web::{web::Path, HttpResponse, Responder};
+use ferrischat_common::types::{InternalServerErrorJson, NotFoundJson};
 
 /// DELETE /api/v0/users/{user_id}
-pub async fn delete_user(Path(user_id): Path<i64>, _: crate::Authorization) -> impl Responder {
+pub async fn delete_user(req: HttpRequest, auth: crate::Authorization) -> impl Responder {
+    let user_id = get_item_id!(req, "user_id");
+
+    if user_id != auth.0 {
+        HttpResponse::Forbidden().finsh()
```
Comment (tazz4843) [inline ferrischat_webserver/src/users/delete_user.rs:9]: typo: `.finish()`

**Review by tazz4843 — CHANGES_REQUESTED** [review]

**Review by tazz4843 — APPROVED** [review]

File: ferrischat_webserver/src/users/delete_user.rs [hunk 3]
```diff
@@ -1,6 +1,32 @@
...
+use ferrischat_common::types::{InternalServerErrorJson, NotFoundJson};
 
 /// DELETE /api/v0/users/{user_id}
-pub async fn delete_user(Path(user_id): Path<i64>, _: crate::Authorization) -> impl Responder {
-    HttpResponse::NoContent()
+pub async fn delete_user(req: HttpRequest, auth: crate::Authorization) -> impl Responder {
+    let user_id = get_item_id!(req, "user_id");
+
+    if user_id != auth.0 {
+        HttpResponse::Forbidden().finish()
```
Comment (tazz4843) [inline ferrischat_webserver/src/users/delete_user.rs:9]: return this from the fn as well
```rs
return HttpResponse::Forbidden().finish();
```

**Review by tazz4843 — CHANGES_REQUESTED** [review]
missed one thing

File: ferrischat_webserver/src/users/delete_user.rs [hunk 4]
```diff
@@ -1,6 +1,32 @@
...
+    }
+    let bigint_user_id = u128_to_bigdecimal!(user_id);
+
+    let db = get_db_or_fail!();
+    let resp = sqlx::query!(
+        "DELETE FROM users WHERE id = $1 RETURNING (id)",
+        bigint_user_id
+    )
+    .fetch_optional(db)
+    .await?;
```
Comment (tazz4843) [inline ferrischat_webserver/src/users/delete_user.rs:19]: the ? here converts this into Ok(T) from `Result<T, E>` and returns `Err(E)` from the function, so also remove this one
```suggestion
    .await;
```

**Review by tazz4843 — CHANGES_REQUESTED** [review]
sorry, one final thing that i missed, besides this it looks good

**Review by tazz4843 — DISMISSED** [review]
one unused import, i'll get that later in one big batch

**Review by hydrostaticcog — APPROVED** [review]
Works as intended on endeavourOS Linux x86_64 (kernel 5.13.12).

**Pull request merged**